Sources/ComposableArchitecture/Store.swift

```diff
@@ -417,9 +417,36 @@ public final class Store<State, Action> {
         }
       case let .run(priority, operation):
         tasks.wrappedValue.append(
-          Task(priority: priority) {
+          Task(priority: priority) { @MainActor in
```

Comment:
Do we really want to mark it as `@MainActor`?

Reply:
We need the reads and writes to `isCompleted` to be serialized, and since the `Send` closure is `@MainActor` it's sufficient to make this one `@MainActor` too for that purpose. This doesn't mean that `operation` will run on the MainActor, but rather just the synchronous parts of the task (i.e. the `isCompleted = true`). We're already doing a MainActor hop in the Publisher case so this isn't straying from established practice.